Gruntfile.js

```diff
@@ -89,56 +231,101 @@ module.exports = function(grunt) {
+      },
+      local: {
+        src: ['dist/local_config.js', 'dist/local_*.js', '!dist/local_autoload.js'],
+        dest: 'build/local.js'
+      },
+      build: {
+        src: ['build/*.js', '!build/main.js', '!build/screeps-profiler.js'],
+        dest: 'build/prepare.js'
+      },
+      main: {
+        src: ['build/prepare.js', 'build/main.js'],
+        dest: 'default/main.js',
+        nonull: true
       }
     },
     copy: {
       main: {
         files: [{
-          expand: true,
-          cwd: 'src/',
-          src: [
-            '**',
-            '!main.js',
-            '!require.js',
-            '!config.js',
-            '!config_logging.js',
-            '!config_brain_memory.js',
-            '!config_brain_nextroom.js',
-            '!config_brain_squadmanager.js',
-            '!config_creep.js',
-            '!config_creep_resources.js',
-            '!config_creep_fight.js',
-            '!config_creep_harvest.js',
-            '!config_creep_mineral.js',
-            //            '!config_creep_move.js',
-            '!config_creep_routing.js',
-            //            '!config_creep_startup_tasks.js',
-            '!config_roomPosition_structures.js',
-            '!config_room.js',
-            '!config_room_basebuilder.js',
-            '!config_room_controller.js',
-            '!config_room_defense.js',
-            '!config_room_market.js',
-            '!config_room_mineral.js',
-            '!config_room_not_mine.js',
-            '!config_room_external.js',
-            '!config_room_flags.js',
-            '!config_room_routing.js',
-            '!config_room_wallsetter.js',
-            '!config_string.js'
-          ],
-          dest: 'dist/',
-        }, {
-          expand: true,
-          cwd: 'node_modules/screeps-profiler',
-          src: ['screeps-profiler.js'],
-          dest: 'dist/'
-        }, {
-          expand: true,
-          cwd: 'screeps-elk/js',
-          src: ['utils.logger.js'],
-          dest: 'dist/'
-        }]
+            expand: true,
+            cwd: 'src/',
+            // TODO add private files later
+            src: ['*.js'],
+            dest: 'dist/'
+          }, {
+            expand: true,
+            cwd: 'src/local/',
+            // TODO add private files later
+            src: ['*.js', '!*.sample'],
+            dest: 'dist/'
+          }, {
+            expand: true,
+            cwd: 'node_modules/screeps-profiler',
+            src: ['screeps-profiler.js'],
+            dest: 'dist/'
+          }, {
+            expand: true,
+            cwd: 'node_modules/screeps-profiler/',
+            src: [
+              'screeps-profiler.js'
+            ],
+            dest: 'dist/'
+          }, {
+            expand: true,
+            cwd: 'node_modules/screeps-profiler/',
+            src: [
+              'screeps-profiler.js'
+            ],
+            dest: 'build/'
+          }
+          // TODO add docs for screeps-elk!
```

Comment:
i ve got n elastic+kibana, would be nice to implement an screeps log shipper to elastic...